src/app/public/modules/text-expand/text-expand.component.ts

```diff
@@ -61,148 +62,176 @@ export class SkyTextExpandComponent implements AfterContentInit {
-          ]
-        );
-      }
+      this.launchModal();
     } else {
-      // Normal View
-      if (!this.isExpanded) {
-        this.setContainerMaxHeight();
-        setTimeout(() => {
-          this.isExpanded = true;
-          this
-            .animateText(this.collapsedText, this.expandedText, true);
-        }, 10);
-
-      } else {
-        this.setContainerMaxHeight();
-        setTimeout(() => {
-          this.isExpanded = false;
-          this
-            .animateText(this.expandedText, this.collapsedText, false);
-        }, 10);
-      }
+      this.toggleText();
     }
   }
 
-  public animationEnd(): void {
-    // Ensure the correct text is displayed
-    this.textExpandAdapter.setText(this.textEl, this.textToShow);
-    // Set height back to auto so the browser can change the height as needed with window changes
-    this.textExpandAdapter.setContainerHeight(this.containerEl, undefined);
-  }
+  public onTransitionEnd(): void {
+    this.disabled = false;
 
-  public ngAfterContentInit(): void {
-    Observable.forkJoin(this.resources.getString('skyux_text_expand_see_more'),
-      this.resources.getString('skyux_text_expand_see_less')).take(1).subscribe(resources => {
-        this.seeMoreText = resources[0];
-        this.seeLessText = resources[1];
-        this.setup(this.expandedText);
-
-        if (!this.expandModalTitle) {
-          this.resources.getString('skyux_text_expand_modal_title').take(1).subscribe(resource => {
-            this.expandModalTitle = resource;
-          });
-        }
-      });
+    // Set the truncated text after the animation has completed.
+    if (!this.isExpanded) {
+      this.textForDisplay = this.getTruncatedText(this.text, this.maxLength);
+      this.textExpandAdapter.setText(this.textElementRef, this.textForDisplay);
+    }
+
+    // Clear out any height styles that were applied during the animation.
+    this.textExpandAdapter.setContainerHeight(this.containerElementRef, undefined);
+    this.changeDetector.markForCheck();
   }
 
-  private setContainerMaxHeight(): void {
-    // ensure everything is reset
-    this.animationEnd();
-    /* Before animation is kicked off, ensure that a maxHeight exists */
-    /* Once we have support for angular v4 animations with parameters we can use that instead */
-    let currentHeight = this.textExpandAdapter.getContainerHeight(this.containerEl);
-    this.textExpandAdapter.setContainerHeight(this.containerEl, `${currentHeight}px`);
+  private reset(): void {
+    const truncatedText = this.getTruncatedText(this.text, this.maxLength);
+    this.textExpandAdapter.setText(this.textElementRef, truncatedText);
+    this.textExpandAdapter.setContainerHeight(this.containerElementRef, undefined);
+
+    this.isExpandable = (truncatedText !== this.text);
+    this.isExpanded = false;
+
+    this.isModal = (
+      this.getNewlineCount(this.text) > this.maxExpandedNewlines ||
+      this.text.length > this.maxExpandedLength
+    );
```

Comment:
Not a big deal, but is this logic a little beefy? Could we encapsulate it with a nicely named method name to explain better what we're comparing here?

Reply:
Sure thing.